application/modules/front/controllers/CollectionController.php

```diff
@@ -173,15 +173,19 @@ public function actionDispbook($collectionName, $ourBookID, $hadithNumbers = NUL
             if ($this->_book->status > 3 and count($pairs) === 1) { // If it's a single-hadith view page
                 $urn = $this->_entries[0][$pairs[0][0]]->englishURN;
                 $nextURN = $this->util->getNextURNInCollection($urn);
-                $previousURN = $this->util->getPreviousURNInCollection($urn);
+				$previousURN = $this->util->getPreviousURNInCollection($urn);
+				$permalinkCanonical = $this->util->get_permalink($this->_entries[1][$pairs[0][1]]->arabicURN, "arabic");
+				
                 if (!is_null($nextURN)) {
                     $viewVars['nextPermalink'] = $this->util->get_permalink($nextURN, "english");
                     $viewVars['nextHadithNumber'] = $this->util->getVerifiedHadithNumber($nextURN, $language = "english");
                 }
                 if (!is_null($previousURN)) {
                     $viewVars['previousPermalink'] = $this->util->get_permalink($previousURN, "english");
                     $viewVars['previousHadithNumber'] = $this->util->getVerifiedHadithNumber($previousURN, $language = "english");
-                }
+				}
+				if ($permalinkCanonical)
```

Comment:
Can you add the condition from before here? `$this->view->params['_pageType'] == "hadith"`? We only want the canonical link to show on hadith pages (sometimes a book might have just one hadith) so it'll satisfy the other criterion.

Reply:
Added a commit with the suggested edits.